frontend/app/components/context-menus/context-menu.service.ts

```diff
@@ -54,21 +57,21 @@ export class ContextMenuService {
 
   // Return the active context menu, if any
   public get active():ContextMenu|null {
-    return this._active_menu;
+    return this.active_menu;
   }
 
-  public close(disableFocus:boolean = false) {
+  public close(disableFocus:boolean = false):Promise<void> {
     if (!this.active) {
-      return this.$q.when(true);
+      return this.$q.when(void 0);
```

Comment:
I know, I am nitpicking here :-) I simply find `void 0` to be uncommon, at least in our code base. Since you can't override `undefined` anyway, I always use `undefined`. Also, TypeScript's `strictNullChecks` check for `undefined`. Not sure it understands `void 0`.